tools/scripts/generate_docs_params.go

```diff
@@ -0,0 +1,198 @@
+
+:::warning
+DO NOT EDIT: this file was generated by make docs_update_gov_params_page
+:::
+
+- [Updating this page](#updating-this-page)
+- [Adding a new parameter](#adding-a-new-parameter)
+- [Parameters](#parameters)
+
+## Updating this page
+
+%s
+make docs_update_gov_params_page
+%s
+
+## Adding a new parameter
+
+Please follow the instructions in [this guide](../../develop/developer_guide/adding_params.md) to add a new parameter.
+
+## Parameters
+
+
+|Module | Field Type | Field Name |Comment |
+|-------|------------|------------|--------|
+`
+
+// writeContentToFile writes the given content to the specified file.
+func writeContentToFile(file_path, content string) error {
+	file, err := os.Create(file_path)
+	if err != nil {
+		return err
+	}
+	defer file.Close()
+
+	// Write the string to the file
+	_, err = file.WriteString(content)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// findProtoFiles returns a slice of file paths that contain the specified pattern within the given base directory.
+func findProtoFiles(baseDir, pattern string) (protoFilePaths []string, err error) {
+	err = filepath.Walk(baseDir, func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+
+		// Check if the file name contains the specified pattern
+		if !info.IsDir() && strings.Contains(info.Name(), pattern) {
+			protoFilePaths = append(protoFilePaths, path)
+		}
+		return nil
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	return protoFilePaths, nil
+}
+
+// prepareGovernanceParamsDocs parses the given .proto files and prepares the governance parameters documentation.
+func prepareGovernanceParamsDocs(protoFilePaths []string, template string) (string, error) {
+	docs := template
+	for _, filePath := range protoFilePaths {
+		fmt.Println("Parsing .proto file:", filePath)
+		module := strings.Split(filePath, "/")[2]
+
+		protoFile, err := os.Open(filePath)
+		if err != nil {
+			fmt.Println("Error opening .proto file:", err)
+			return "", err
+		}
+		defer protoFile.Close()
+
+		var paramsMessages []ProtoMessage
+		var currentParamMessage *ProtoMessage
+		var currentComment string
+
+		messageParamPattern := regexp.MustCompile(`^message\s+(Params)\s*{`)
+		fieldPattern := regexp.MustCompile(`^\s*(\w+)\s+(\w+)\s*=\s*(\d+)\s*\[(.*?)\];`)
+		commentPattern := regexp.MustCompile(`^//\s*(.*)`)
+
+		scanner := bufio.NewScanner(protoFile)
+		for scanner.Scan() {
+			line := strings.TrimSpace(scanner.Text())
+
+			// Check if the line is a comment
+			if matches := commentPattern.FindStringSubmatch(line); matches != nil {
+				currentComment += matches[1] + " "
+				continue
+			}
+
+			// Check if the line defines a new message
+			if matches := messageParamPattern.FindStringSubmatch(line); matches != nil {
+				if currentParamMessage != nil {
+					paramsMessages = append(paramsMessages, *currentParamMessage)
+				}
+				currentParamMessage = &ProtoMessage{Name: matches[1]}
+				continue
```

Comment:
We should reset any comment seen before seeing `message Params` for the first time.

Reply:
This caught a bug!